core/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -206,7 +207,8 @@ public boolean hasNext()
                     }
                   }
                   if (currentBatch == null) {
-                    throw new QueryTimeoutException("Sequence iterator timed out waiting for data");
+                    throw new QueryTimeoutException("Query did not complete within configured timeout period. " +
```

Comment:
Suggestion: if the error appears more than once, declare a constant and reference the constant. That also makes it easier or tests to track the message as it evolves.

Reply:
👍